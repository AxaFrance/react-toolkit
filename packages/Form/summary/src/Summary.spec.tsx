import React from 'react';
import renderer from 'react-test-renderer';
import { Summary } from './index';

describe('<Summary>', () => {
  it('Renders correctly with default props', () => {
    const tree = renderer.create(<Summary />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly when not visible', () => {
    const tree = renderer.create(<Summary isVisible={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly when visible', () => {
    const messages = ['Sample message 1', '', 'Sample message 2', null];
    const tree = renderer.create(<Summary messages={messages} isVisible />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
