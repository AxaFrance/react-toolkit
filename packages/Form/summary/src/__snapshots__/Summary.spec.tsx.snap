// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary> Renders correctly when not visible 1`] = `null`;

exports[`<Summary> Renders correctly when visible 1`] = `
<div
  className="af-alert af-alert--danger"
  role="alert"
>
  <div
    className="af-alert__title"
  >
    <div
      className="af-alert__title-icon"
    >
      <i
        className="glyphicon glyphicon-warning-sign"
      />
    </div>
    <div
      className="af-alert__title-text"
    >
      Invalid form
    </div>
  </div>
  <div
    className="af-alert__content"
  >
    <div
      className="af-alert__content__left"
    />
    <div
      className="af-alert__content__right"
    >
      <ul
        className="af-summary__message-list"
      >
        <li
          className="af-summary__message-item"
        >
          <span>
            Sample message 1
          </span>
        </li>
        <li
          className="af-summary__message-item"
        >
          <span>
            Sample message 2
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Summary> Renders correctly with default props 1`] = `null`;
